homeassistant/components/frontier_silicon/browse_media.py

```diff
@@ -0,0 +1,242 @@
+"""Support for media browsing."""
+import asyncio
+import logging
+
+from homeassistant.components.media_player import BrowseError, BrowseMedia
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_CHANNEL,
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_TYPE_CHANNEL,
+)
+
+CHILD_TYPE_MEDIA_CLASS = {
+    MEDIA_TYPE_CHANNEL: MEDIA_CLASS_CHANNEL,
+    "preset": MEDIA_CLASS_CHANNEL,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class UnknownMediaType(BrowseError):
+    """Unknown media type."""
+
+
+async def build_item_response(media_library, payload, get_thumbnail_url=None):
+    """Create response payload for the provided media query."""
+    search_id = payload["search_id"]
+    search_type = payload["search_type"]
+
+    _, title, media = await get_media_info(media_library, search_id, search_type)
+    if get_thumbnail_url is not None:
+        thumbnail = await get_thumbnail_url(search_type, search_id)
+    else:
+        thumbnail = None
+
+    if media is None:
+        return None
+
+    children = await asyncio.gather(
+        *(item_payload(item, get_thumbnail_url) for item in media)
+    )
+
+    response = BrowseMedia(
+        media_class=MEDIA_CLASS_DIRECTORY,
+        media_content_id=search_id,
+        media_content_type=search_type,
+        title=title,
+        can_play=bool(search_id),
+        can_expand=True,
+        children=children,
+        thumbnail=thumbnail,
+    )
+    response.calculate_children_class()
+
+    return response
+
+
+async def item_payload(item, get_thumbnail_url=None):
+    """
+    Create response payload for a single media item.
+
+    Used by async_browse_media.
+    """
+    title = item["label"]
+
+    media_class = None
+
+    if "key" in item:
+        media_content_type = f"{item['type']}"
+        media_content_id = f"{item['key']}"
```

Comment:
No need to wrap in f-string.
```suggestion
        media_content_type = item['type']
        media_content_id = item['key']
```